src/main/java/net/fabricmc/loader/entrypoint/minecraft/hooks/EntrypointUtils.java

```diff
@@ -17,38 +17,47 @@
 package net.fabricmc.loader.entrypoint.minecraft.hooks;
 
 import net.fabricmc.loader.FabricLoader;
-import net.fabricmc.loader.gui.FabricGuiEntry;
 
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
 import java.util.function.Consumer;
 
-final class EntrypointUtils {
-	private EntrypointUtils() {
+public final class EntrypointUtils {
+	public static <T> void invoke(String name, Class<T> type, Consumer<? super T> invoker) {
```

Comment:
Can we change the generic on class to `Class<? extends T>` just in case we have generic entry points in the future (e.g. a concrete type that can be cast to `Entry<? extends MinecraftServer>`)

Reply:
I'm not sure that's the right fix or necessary, type is usually supplied in erased form. We can change it if the need arises since this is no public API.